### ISSUE TITLE
forked-process-task-runner doesn't recognize "clear line" ESC sequence when injecting prefix

### ISSUE THREAD

**Issue by dko-slapdash** [nrwl/nx#15586]
### Current Behavior

TL;DR: below is a patch which fixes lerna and nx colored output prefixing behavior when the underlying package manager (e.g. yarn) emits "clear line" ESC code.

The usecase is when forked-process-task-runner is used from inside lerna.

When NX_PREFIX_OUTPUT=true and streamOutput is active, forkProcessPipeOutputCapture() injects `task.target.project` as a prefix to each line of stdout/stderr. Some package managers though (e.g. yarn) use special ESC characters in their output, so they emit "clear line, move cursor to column 1", and this obviously erases the prefix which nx injects:

![CleanShot 2023-03-10 at 01 53 18@2x](https://user-images.githubusercontent.com/46383452/224287242-1134c682-b7f1-4d83-a447-48445884a514.png)

Notice that `$ lint.sh` on the screenshot above doesn't have a prefix. It **should've been prefixed**, but it's not, because yarn emitted `\x1b[2K\x1b\[1G...$ lint.sh` in practice (where `\x1b[2K` is "clear line" and `\x1b\[1G` is "move cursor to column 1").

### Expected Behavior

This is how the output should've looked (notice that `$ lint.sh` is prefixed):

<img width="728" alt="CleanShot 2023-03-10 at 01 52 29@2x" src="https://user-images.githubusercontent.com/46383452/224286236-3ac0f3df-f12c-445e-bf43-8de419408d28.png">

### Steps to Reproduce

In this exampke, 

```
yarn -s lerna run --scope=slapdash-shared --concurrency=1 --stream lint
```

### Nx Report

```shell
Irrelevant; it's lerna related (which uses nx internal libs).
```

### Additional Information

Here is a patch-package which solves this issue. Hope you folks will be able to merge this approach into the mainstream code.

```
diff --git a/node_modules/nx/src/tasks-runner/forked-process-task-runner.js b/node_modules/nx/src/tasks-runner/forked-process-task-runner.js
index f1e444b..120e4eb 100644
--- a/node_modules/nx/src/tasks-runner/forked-process-task-runner.js
+++ b/node_modules/nx/src/tasks-runner/forked-process-task-runner.js
@@ -110,9 +110,11 @@ class ForkedProcessTaskRunner {
                         const color = getColor(task.target.project);
                         const prefixText = `${task.target.project}:`;
                         p.stdout
+                            .pipe(logClearLineToPrefixTransformer(color.bold(prefixText) + " "))
                             .pipe(logTransformer({ tag: color.bold(prefixText) }))
                             .pipe(process.stdout);
                         p.stderr
+                            .pipe(logClearLineToPrefixTransformer(color(prefixText) + " "))
                             .pipe(logTransformer({ tag: color(prefixText) }))
                             .pipe(process.stderr);
                     }
@@ -350,4 +352,17 @@ function getColor(projectName) {
     const colorIndex = code % colors.length;
     return colors[colorIndex];
 }
+function logClearLineToPrefixTransformer(prefix) {
+    let prevChunk = null;
+    return new require("stream").Transform({
+        transform(chunk, _encoding, callback) {
+            if (prevChunk && prevChunk.toString() === "\x1b[2K") {
+                chunk = chunk.toString().replace(/\x1b\[1G/g, (m) => m + prefix);
+            }
+            this.push(chunk);
+            prevChunk = chunk;
+            callback();
+        },
+    });
+} 
 //# sourceMappingURL=forked-process-task-runner.js.map
\ No newline at end of file
```
